app/main.py

```diff
@@ -1,2 +1,100 @@
 class Dictionary:
-    pass
+    def __init__(self):
+        self._hash_table_size = 8
+        self._cells = [None] * self._hash_table_size
+
+    @property
+    def table(self):
+        table = []
+        for cell in self._cells:
+            if cell is not None:
+                key, _, value = cell
+                table.append((key, value))
+
+        return table
+
+    def __setitem__(self, key, value):
+        if self._is_necessary_hash_table_resizing():
+            self._perform_hash_table_resize()
+
+        self._fill_hash_table_cell(self._cells, key, value)
+
+    def __getitem__(self, item):
+        hashed_index = self._get_hashed_index(item)
+
+        while self._cells[hashed_index] is not None:
+            cell_key, cell_hash, cell_value = self._cells[hashed_index]
+            if cell_key == item and cell_hash == hash(item):
+                return cell_value
+            hashed_index = (hashed_index + 1) % self._hash_table_size
+        else:
+            raise KeyError
+
+    def __len__(self):
+        return sum((cell is not None for cell in self._cells))
```

Comment:
you should create attribute length and use it. Your solution it's additional time for working dictionary

Reply:
I couldn't imagine that changing this method could make the program 5x faster